src/fraction.ts

```diff
@@ -64,10 +67,35 @@ export class Fraction {
   }
 
   toNumber() {
-    return (
-      parseInt(this.numerator.toString()) /
-      parseInt(this.denominator.toString())
+    let numerator = this.numerator.clone();
+    let denominator = this.denominator.clone();
```

Comment:
Does `BN` not implement division after which we can convert to number? It is surprising to me that library wouldn't but I haven't checked myself.

Reply:
It implements integer division, so we would lose fractional precision in some cases